libsel4vm/src/arch/x86/processor/msr.c

```diff
@@ -20,17 +20,44 @@
 #include "processor/lapic.h"
 #include "interrupt.h"
 
+#ifdef CONFIG_ARCH_X86_64
+static seL4_Word vm_msr_read(seL4_CPtr vcpu, unsigned int field)
+{
+
+    seL4_X86_VCPU_ReadMSR_t result;
+
+    assert(vcpu);
+
+    result = seL4_X86_VCPU_ReadMSR(vcpu, (seL4_Word)field);
+    assert(result.error == seL4_NoError);
+
+    return result.value;
+}
+
+static void vm_msr_write(seL4_CPtr vcpu, unsigned int field, seL4_Word value)
+{
+
+    seL4_X86_VCPU_WriteMSR_t result;
+
+    assert(vcpu);
+
+    result = seL4_X86_VCPU_WriteMSR(vcpu, (seL4_Word)field, value);
+    assert(result.error == seL4_NoError);
```

Comment:
Doesn't this complain that the value assigned to `result` is unused in release builds?
Consider using this:
```
result = ...
ZF_LOGF_IF(result.error != result, "writing failes, result = ...");
```
which always calls `abort()` then the condition fails. Otherwise declare declare `seL4_X86_VCPU_WriteMSR_t UNUSED result;` or use something like this:
```
if (result.error != result {
    ZF_LOGE("writing failed, result = ...");
    assert(0);
```